### ISSUE TITLE
Add a way to treat empty output as a failure to slugify

### ISSUE THREAD

**Issue by dmbaturin** [thangngoc89/ocaml-slug#7]
Many scripts are not supported and many would be nearly impossible to support in a small library. Right now `Slug.slugify` simply returns an empty string when it sees a script it doesn't support.

```
utop # Slug.slugify " देवनागरी";;
- : string = ""

utop # Slug.slugify "漢字";;
- : string = ""
```

However, an empty string (or a string with most characters removed) isn't a useful slug. Library users should be able to know how well the slugification went so that they can fall back to an alternative method.

I believe `Slug.slugify` should raise an `Invalid_argument` exceptions when a string contains unsupported script. Maybe there should also be `slugify_unsafe` that ignores such errors.

**Comment by thangngoc89** [comment]
@dmbaturin maybe unsupported could be replaced by a character like `?` because I don't want to throw errors in this case. 

The related line is here https://github.com/thangngoc89/ocaml-slug/blob/0725508c60e3a44666251d3e421c457c95ee09dc/slug.ml#L30 

Are you interested in sending a PR that allow `~by:""` to be configurable?

**Comment by dmbaturin** [comment]
Well, suppose someone who writes exclusively in Japanese or Hindi installs a program that uses the lib. They will end up with every slug neither readable nor unique, and they will not know until they inspect the output visually.

I believe it's better to reject unsupported inputs outright.

**Comment by thangngoc89** [comment]
@dmbaturin I would be happy to accept `Slug.slugify_exn` that raise `Invalid_argument` when there are non_alphaNum characters. I really don't want to throw in the main function.

**Comment by joprice** [comment]
How about returning a `(string, error) result` instead, where error has some data about what went wrong?

**Comment by thangngoc89** [comment]
@joprice Yes. That is acceptable. I’m willing to accept any PRs regardings this but I won’t spend anytime doing this myself since I have no use case for such feature.
